pandas/io/sas/sas7bdat.py

```diff
@@ -51,18 +63,10 @@ def _convert_datetimes(sas_datetimes: pd.Series, unit: str) -> pd.Series:
     try:
         return pd.to_datetime(sas_datetimes, unit=unit, origin="1960-01-01")
     except OutOfBoundsDatetime:
-        if unit == "s":
-            s_series = sas_datetimes.apply(
-                lambda sas_float: datetime(1960, 1, 1) + timedelta(seconds=sas_float)
-            )
+        if unit in ["s", "d"]:
```

Comment:
wondering if its better to move the arg check, and `raise` into the `_parse_datetime()` function?

Reply:
yes, makes sense, that way we don't double check. Will do